src/operator/aggregate/mod.rs

```diff
@@ -75,9 +105,20 @@ pub trait Aggregator<K, T, R>: Clone + 'static {
     ///
     /// * The method must return `None` if the total weight of each key is zero.
     ///   It must return `Some` otherwise.
-    fn aggregate<'s, C>(&self, cursor: &mut C) -> Option<Self::Output>
+    fn aggregate<'s, C>(&self, cursor: &mut C) -> Option<Self::Accumulator>
     where
         C: Cursor<'s, K, (), T, R>;
+
+    /// Compute the final value of the aggregate.
+    fn finalize(&self, accumulator: Self::Accumulator) -> Self::Output;
+
+    /// Applies `aggregate` to `cursor` followed by `finalize`.
+    fn aggregate_finalize<'s, C>(&self, cursor: &mut C) -> Option<Self::Output>
```

Comment:
this name is misleading, it suggests that only the finalize method of the aggregator is applied.
it should be called "aggregate_and_finalize"